mail_queue_job/models/mail_mail.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import psycopg2
+from odoo import api, fields, models
+from odoo.addons.queue_job.job import job, DEFAULT_PRIORITY
+
+
+class MailMail(models.Model):
+    _inherit = 'mail.mail'
+
+    mail_job_priority = fields.Integer(
+        oldname='priority',
+        default=DEFAULT_PRIORITY,
+        help="Send the email with a priority level.\n"
+             "0 being the higher priority. Default is %d" % DEFAULT_PRIORITY,
+    )
+
+    @api.multi
+    @job(default_channel='root.mail')
+    def _send_mail_jobified(self):
+        """
+        Send the email if
+        - the state is 'outgoing'
+        - the mail is not processing by another job (locked)
+        - the mail exists.
+        :return: str
+        """
+        # Make sure two jobs don't send the same email.
```

Comment:
@acsonefho It would be nice to forward port https://github.com/OCA/queue/pull/66 in 11.0 To be able to use the new concept of Idendity key on queue job. This forward port could solve the problem of multiple jobs for the same email in a more 'elegant' way.